### PR TITLE
matrix-puppet-discord: init at 0.0.0

### PR DESCRIPTION
**Pull request by bqv**
If you like.

### CODE REVIEW

File: pkgs/public/matrix/matrix-puppet-discord/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+let
+  nodePackages = import ./node-composition.nix {
+    inherit pkgs;
+    inherit (stdenv.hostPlatform) system;
+    inherit nodejs;
+  };
+in stdenv.mkDerivation rec {
+  pname = "matrix-puppet-discord";
+  version = "0.0.0";
```
Comment (arcnmx) [inline pkgs/public/matrix/matrix-puppet-discord/default.nix]: This should probably match typical unstable dated version names if it's not a tagged release.
```suggestion
  version = "2020-03-04";
```

File: pkgs/public/matrix/default.nix [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   matrix-puppet-skype = import ./matrix-puppet-skype;
+  matrix-puppet-discord = import ./matrix-puppet-discord;
```
Comment (arcnmx) [inline pkgs/public/matrix/default.nix]: I think this should be `mx-puppet-discord` instead? It doesn't match the others but is the project's name, and the `matrix-puppet-X` prefix is mostly taken by someone else's projects.

**Comment by arcnmx** [comment]
Thanks for this, I've been meaning to try the bridge out myself some point but hadn't gotten around to it yet. How well is it working for you?

I'm happy to merge and can fix some of it up after, but do feel that the npm deps should probably match the repo's lockfile.

File: pkgs/public/matrix/matrix-puppet-discord/default.nix (hunk 1 above)
Comment (bqv) [reply pkgs/public/matrix/matrix-puppet-discord/default.nix]: Good shout

File: pkgs/public/matrix/default.nix (hunk 2 above)
Comment (bqv) [reply pkgs/public/matrix/default.nix]: Ah, didn't realise that. Done :)

**Comment by bqv** [comment]
No worries! I've amended that, should match now. :+1:

**Comment by arcnmx** [comment]
Okay so this is a little nitpicky but one final request: could you include something like c3cca1d540d0489dbcae31c5ad8e2130c61f7751? It just feels a bit wasteful to include the json because of a quirk in node2nix.

**Comment by bqv** [comment]
Gotcha, hopefully that's okay. I agree but it does make it easier to upgrade later if deps change. Fair to assume they won't though :D

**Comment by arcnmx** [comment]
Yeah the problem was more that the derivation was depending on `./.`, so it was pulling in even `default.nix` and unrelated nix expressions - things like changing metadata would cause rebuilds. Plus committing 2000 lines of unused json just feels wasteful.

Anyway, thanks a lot for everything, looks great! Hopefully I can get around to trying it out myself soon.

**Pull request merged**